imageio/imageio-jpeg/src/main/java/com/twelvemonkeys/imageio/plugins/jpeg/JPEGImageReader.java

```diff
@@ -289,70 +306,203 @@ public ImageTypeSpecifier getRawImageType(int imageIndex) throws IOException {
 
     @Override
     public BufferedImage read(int imageIndex, ImageReadParam param) throws IOException {
-        checkBounds(imageIndex);
-        initHeader(imageIndex);
+        return rotateIfNecessary(readImage(imageIndex, param));
```

Comment:
Jup, the destination image would not work at this point, you are absolutely right. I will have to find another solution for this. As suggested by you, we should perhaps actually avoid the delegate in this case.

Reply:
I think maybe we could use a similar strategy as you use for the source region... Perhaps clear the destination in the param (or possibly better, copy the param with selected values) before passing to the delegate, and then copy the result into the destination in the correct orientation. 

This will work, in theory, but it is costly in terms of memory and excessive copying as we need twice the amount of memory to decode the image. However, we already do this in the `readImageAsRasterAndReplaceColorProfile` (to apply correct color conversion, icc profile etc), so it may not be much of an issue, as long as we always chose that code path for orientation > 1.